python/lsst/daf/butler/core/storageClass.py

```diff
@@ -272,9 +289,56 @@ def addFromConfig(self, config):
             storageClassKwargs["components"] = components
 
             # Create the new storage class and register it
-            newStorageClass = StorageClass(name, **storageClassKwargs)
+            baseClass = StorageClass
+            if "inheritsFrom" in info:
+                baseName = info["inheritsFrom"]
+                if baseName not in self:
+                    processStorageClass(baseName, sconfig)
+                baseClass = type(self.getStorageClass(baseName))
+
+            newStorageClassType = self.makeNewStorageClass(name, baseClass, **storageClassKwargs)
+            newStorageClass = newStorageClassType()
             self.registerStorageClass(newStorageClass)
 
+        for name in list(sconfig.keys()):
+            processStorageClass(name, sconfig)
+
+    @staticmethod
+    def makeNewStorageClass(name, baseClass, **kwargs):
+        """Create a new Python class as a subclass of `StorageClass`.
+
+        Parameters
+        ----------
+        name : `str`
+            Name to use for this class.
+        baseClass : `type`
+            Base class for this `StorageClass`.
```

Comment:
Should this argument have a default, or is this method never used for StorageClasses that do not have a base class?

Reply:
I'll add a default. It's questionable whether this should be a public method at all.